src/main/java/duckling/responders/FolderContents.java

```diff
@@ -40,18 +40,23 @@ public boolean matches() {
     public ArrayList<String> headers() throws IOException {
         ResponseHeaders headers = new ResponseHeaders();
 
-        if (methodNotAllowed()) return headers.notAllowed().toList();
+        if (!isAllowed()) return headers.notAllowed().toList();
 
         return headers.withContentType("text/html").toList();
     }
 
     @Override
     public InputStream body() throws IOException {
-        String responseContent = methodNotAllowed() ? "" : rawFolderResponse();
+        String responseContent = !isAllowed() ? "" : rawFolderResponse();
 
         return new ByteArrayInputStream(responseContent.getBytes());
     }
 
+    @Override
+    public boolean isAllowed() {
+        return this.allowedMethods.contains(request.getMethod());
+    }
+
```

Comment:
This implementation looks the same as in `FileContents` and `DefinedContents`. I wonder if it makes sense to centralize this logic somewhere.